components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -210,6 +211,37 @@ class BrowserToolbar @JvmOverloads constructor(
             editToolbar.urlView.setTextColor(value)
         }
 
+    /**
+     * Sets the icon of the tracking protection the site has tracking protection enabled, but none
+     * trackers have been blocked or detected.
+     */
+    var trackingProtectionONIcon: Drawable
+        get() = displayToolbar.trackingProtectionONIcon
+        set(value) {
+            displayToolbar.trackingProtectionONIcon = value
+        }
+
+    /**
+     * Sets the icon of the tracking protection icon when is active
+     * e.g (Trackers have been blocked or detected)
+     */
+    var trackingProtectionActiveIcon: Drawable
+        get() = displayToolbar.trackingProtectionActiveIcon
+        set(value) {
+            displayToolbar.trackingProtectionActiveIcon = value
+        }
+
+    /**
+     * Sets the colour of the vertical separator between the tracking protection icon and the
+     * security indicator icon.
+     */
+    @get:ColorInt
+    var trackingProtectionAndSecurityIndicatorSeparatorColor: Int
+        get() = displayToolbar.trackingProtectionAndSecurityIndicatorSeparatorColor
+        set(@ColorInt value) {
+            displayToolbar.trackingProtectionAndSecurityIndicatorSeparatorColor = value
+        }
```

Comment:
Those are a lot of fields. I wonder if we could use a state list drawable instead?

Reply:
Fenix is doing the animations using [lottie](https://github.com/airbnb/lottie-android) to support that  [they have to build programmatically the lottie object and pass us a drawable via code](https://github.com/mozilla-mobile/fenix/pull/4310/files#diff-b9d62bd5f486cff5a941b3c7b84f0fa4R68)

Reply:
Mhhh :( - Could we still build a state list drawable in code from that?

Reply:
I'm not quite sure. I'll take a look!
But at the end of the day we are going to need three icons from the app:

`ON_TRACKERS_BLOCKED`
`ON_NO_BLOCKED_TRACKERS`
`OFF_FOR_THIS_SITE`

Reply:
Yes, but setting one drawable for tracking protection and internally the drawable has three states that we use will be much much nicer as public API :)

Reply:
Agree :)

Reply:
Addressed now we have:

        `ON_NO_TRACKERS_BLOCKED`
        `ON_TRACKERS_BLOCKED`
        `OFF_FOR_A_SITE`
        `OFF_GLOBALLY`